### PR TITLE
Token Endorsements

### PR DESCRIPTION
**Pull request by pcostantini**
Based on C# work (https://github.com/Microsoft/botbuilder-dotnet/pull/655)

### CODE REVIEW

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java [hunk 1]
```diff
@@ -74,8 +74,8 @@
      * @return A valid ClaimsIdentity.
      * @throws AuthenticationException A token issued by the Bot Framework emulator will FAIL this check.
      */
-    public static CompletableFuture<ClaimsIdentity> authenticateToken(String authHeader,CredentialProvider credentials, String serviceUrl) throws ExecutionException, InterruptedException, AuthenticationException {
-        ClaimsIdentity identity = ChannelValidation.authenticateToken(authHeader, credentials).get();
+    public static CompletableFuture<ClaimsIdentity> authenticateToken(String authHeader,CredentialProvider credentials, String channelId, String serviceUrl) throws ExecutionException, InterruptedException, AuthenticationException {
```
Comment (cleemullins) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java]: Missing the @param for channelId.

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java [hunk 2]
```diff
@@ -24,13 +24,13 @@
      * @return A valid ClaimsIdentity.
      * @throws AuthenticationException A token issued by the Bot Framework emulator will FAIL this check.
      */
-    public static CompletableFuture<ClaimsIdentity> authenticateToken(String authHeader, CredentialProvider credentials) throws ExecutionException, InterruptedException, AuthenticationException {
+    public static CompletableFuture<ClaimsIdentity> authenticateToken(String authHeader, CredentialProvider credentials, String channelId) throws ExecutionException, InterruptedException, AuthenticationException {
```
Comment (cleemullins) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java]: Missing the @param for channelId

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java (hunk 1 above)
Comment (pcostantini) [reply libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java]: Updated

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java (hunk 2 above)
Comment (pcostantini) [reply libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ChannelValidation.java]: Updated

**Pull request merged**